src/sentry/static/sentry/app/utils/apm.jsx

```diff
@@ -4,6 +4,7 @@ import * as Sentry from '@sentry/browser';
 let firstPageLoad = true;
 let flushTransactionTimeout = undefined;
 let wasInterrupted = false;
+let currentTransactionSpan = null;
```

Comment:
@billyvg I think you may need to set `currentTransactionSpan` to be `null` in the `finishTransaction()` function for the timeout case.

Reply:
@dashed yes and no - you can only call `finish()` on a span once, so it wouldn't do anything the next time you called `startTransaction()`, we should reset it though since it can cause unexpected results in future though.